### DISCUSSION TITLE
Behavior on Build

### DISCUSSION THREAD

**Discussion by noel** [AltimateAI/vscode-dbt-power-user#39] [Q&A] [answered]
Did something change with the latest release?  When I used to build a model, the command went to the open terminal window and there I would see success/failure.  In the latest version (0.4.3) I am switched to the output tab.  Is there a way to disable this?

**Comment by mdesmet** [comment, marked as answer]
We now depend on the Python extension to better support virtual environments. See https://code.visualstudio.com/docs/languages/python#_environments. We don't want depend anymore on user terminal, as we will come with more features that depend on DBT internals or other python code.

When you run a model with the play button, it should output the build log there. For now the output channel doesn't support colors, but you should still see the result of your run there. Do you see your DBT version being detected as in the screen shots on the [README](https://github.com/innoverio/vscode-dbt-power-user/blob/master/README.md)?

We will very soon come out will a new version that works just as in the terminal but in a readonly version but still using the principes stated above.

Let us know if it works!

**Reply by noel** [reply]
I see, I am using virtual envs and VS Code automatically applies them to my terminal within VS Code, so this was never a problem.

I do miss the colors so that will help.  What I also liked about the terminal version is that I could take the command and change it.  e.g. up arrow and add a + to run for all dependent models.  The new process would require that I type/copy it after switching tabs.

I am also seeing a ton of output (maybe validations?) in the output window that I was not seeing in terminal, so that is very distracting.

What about a setting that allows the new default or the old behavior?

Btw, I do love the extension so don't want to come across as too demanding or unappreciative.  Thank you for creating and maintaining this.  It is still very helpful.

**Reply by mdesmet** [reply]
There were quite some users telling us that sometimes the command gets executed before the venv was activated. We definitely don't want to put a lot of if/else behaviors in the code, recipe for disaster!

Actually what you confused you is the output of the dbt list command (needed for the autocompletion etc). 

We can simply add buttons for the children and parent models. You can also go to the dbt sidebar menu and execute children and parent models from there.

We just released a new version:
- Run commands through a read only terminal
- Cancellation possible of commands through the cancel button
- Colored output
- Clearer separation between different commands

**Reply by noel** [reply]
oh, I see, I usually had the terminal open so I see how it can be a problem if it was not.

So far the latest version I am getting is 0.4.3 guessing it will take a bit for VS Code to see new version.

Another thing I noticed on the output tab is that it is not scrolling to the end.  e.g. I have to manually scroll to see the latest output.

**Reply by noel** [reply]
I got the latest version.  definitely better.  Is there a reason for the list after each command is run?  I noticed the issue with scroll has to do with me scrolling up a bit to read a msg, after that next commend won't scroll to bottom of output.

**Reply by mdesmet** [reply]
It should only be triggered when a source file has changed. We'll have a look if we can find something fishy!
